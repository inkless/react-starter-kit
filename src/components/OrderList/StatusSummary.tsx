import React from 'react';

import { theme } from '@src/theme';
import Box from '@src/components/base/Grid/Box';
import Text from '@src/components/base/Typography/Text';
import Card from '@src/components/base/Card/Card';

export type Props = {
  statuses: { name: string; count: number }[];
};

/**
 * Render StatusSummary
 */
export function StatusSummary(props: Props) {
  // sum all statuses
  const total = props.statuses.reduce((acc, status) => {
    return acc + status.count;
  }, 0);

  return (
    <Card display="flex" padding={0} marginBottom={3}>
      {props.statuses.map((status, index) => (
        <Box key={index} flex="1 0 0" borderLeft={index && `1px solid ${theme.colors.grays[3]}`}>
          {/* status name */}
          <Text padding={0} fontSize={3} fontWeight="bold" textAlign="center" marginBottom={1}>
            {status.name}
          </Text>

          {/* status percentage */}
          <Text
            padding={0}
            fontSize={3}
            fontWeight="bold"
            textAlign="center"
            color={theme.colors.grays[8]}
            marginBottom={1}
          >
            {Math.round((1000 * status.count) / total) / 10}%
          </Text>

          {/* status order count */}
          <Text padding={0} fontSize={2} textAlign="center" color={theme.colors.grays[5]}>
            {status.count} orders
          </Text>
        </Box>
      ))}
    </Card>
  );
}

export default StatusSummary;
