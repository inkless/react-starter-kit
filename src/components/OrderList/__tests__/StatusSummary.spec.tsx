import React from 'react';
import { shallow } from 'enzyme';

import { statuses } from './__fixture__/statuses';
import StatusSummary from '../StatusSummary';

describe('Component StatusSummary', () => {
  it('should render StatusSummary correctly', () => {
    const component = shallow(<StatusSummary statuses={statuses} />);

    expect(component).toMatchSnapshot();
  });
});
