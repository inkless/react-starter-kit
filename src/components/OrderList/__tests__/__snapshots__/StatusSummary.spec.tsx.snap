// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component StatusSummary should render StatusSummary correctly 1`] = `
<Styled(styled.div)
  borderRadius={2}
  boxShadow={1}
  display="flex"
  marginBottom={3}
  padding={0}
>
  <styled.div
    borderLeft={0}
    flex="1 0 0"
    key="0"
  >
    <Styled(styled.div)
      color="#484848"
      fontSize={3}
      fontWeight="bold"
      marginBottom={1}
      padding={0}
      textAlign="center"
    >
      Pending
    </Styled(styled.div)>
    <Styled(styled.div)
      color="#656565"
      fontSize={3}
      fontWeight="bold"
      marginBottom={1}
      padding={0}
      textAlign="center"
    >
      16.7
      %
    </Styled(styled.div)>
    <Styled(styled.div)
      color="#9b9b9b"
      fontSize={2}
      padding={0}
      textAlign="center"
    >
      21
       orders
    </Styled(styled.div)>
  </styled.div>
  <styled.div
    borderLeft="1px solid #c3c3c3"
    flex="1 0 0"
    key="1"
  >
    <Styled(styled.div)
      color="#484848"
      fontSize={3}
      fontWeight="bold"
      marginBottom={1}
      padding={0}
      textAlign="center"
    >
      In Route
    </Styled(styled.div)>
    <Styled(styled.div)
      color="#656565"
      fontSize={3}
      fontWeight="bold"
      marginBottom={1}
      padding={0}
      textAlign="center"
    >
      33.3
      %
    </Styled(styled.div)>
    <Styled(styled.div)
      color="#9b9b9b"
      fontSize={2}
      padding={0}
      textAlign="center"
    >
      42
       orders
    </Styled(styled.div)>
  </styled.div>
  <styled.div
    borderLeft="1px solid #c3c3c3"
    flex="1 0 0"
    key="2"
  >
    <Styled(styled.div)
      color="#484848"
      fontSize={3}
      fontWeight="bold"
      marginBottom={1}
      padding={0}
      textAlign="center"
    >
      Delivered
    </Styled(styled.div)>
    <Styled(styled.div)
      color="#656565"
      fontSize={3}
      fontWeight="bold"
      marginBottom={1}
      padding={0}
      textAlign="center"
    >
      50
      %
    </Styled(styled.div)>
    <Styled(styled.div)
      color="#9b9b9b"
      fontSize={2}
      padding={0}
      textAlign="center"
    >
      63
       orders
    </Styled(styled.div)>
  </styled.div>
</Styled(styled.div)>
`;
